qt/python/mantidqt/mantidqt/widgets/sliceviewer/cutviewer/view.py

```diff
@@ -0,0 +1,317 @@
+        self.setLayout(self.layout)
+
+    def _setup_table_widget(self):
+        """
+        Make a table showing
+        :return: A QTableWidget object which will contain plot widgets
+        """
+        table_widget = QTableWidget(3, 7, self)
+        table_widget.setVerticalHeaderLabels(['u1', 'u2', 'u3'])
+        col_headers = ['a*', 'b*', 'c*'] if self.frame == SpecialCoordinateSystem.HKL else ['Qx', 'Qy', 'Qz']
+        col_headers.extend(['start', 'stop', 'nbins', 'step'])
+        table_widget.setHorizontalHeaderLabels(col_headers)
+        table_widget.setFixedHeight(table_widget.verticalHeader().defaultSectionSize()*(table_widget.rowCount()+1))  # +1 to include headers
+        for icol in range(table_widget.columnCount()):
+            table_widget.setColumnWidth(icol, 50)
+        table_widget.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+        table_widget.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)
+
+        self.table = table_widget
+        self.layout.addWidget(self.table)
+
+    def _setup_figure_widget(self):
+        fig, _, _, _ = create_subplots(1)
+        fig.axes[0].autoscale(enable=True, tight=False)
+        self.figure = fig
+        self.figure.canvas = FigureCanvas(self.figure)
+        toolbar = MantidNavigationToolbar(self.figure.canvas, self)
+        self.figure_layout = QVBoxLayout()
+        self.figure_layout.addWidget(toolbar)
+        self.figure_layout.addWidget(self.figure.canvas)
+        self.layout.addLayout(self.figure_layout)
+
+    def _init_slice_table(self):
+        for icol in range(self.table.columnCount()):
+            for irow in range(self.table.rowCount()):
+                item = QTableWidgetItem()
+                if icol == 5:
+                    item.setData(Qt.EditRole, int(1))
+                else:
+                    item.setData(Qt.EditRole, float(1))
+                if irow == self.table.rowCount()-1:
+                    item.setFlags(item.flags() ^ Qt.ItemIsEditable)  # disable editing in last row (out of plane dim)
+                    item.setBackground(QColor(250, 250, 250))
+                else:
+                    item.setFlags(item.flags() | Qt.ItemIsEditable)
+                self.table.setItem(irow, icol, item)
+
+    def _format_cut_figure(self):
+        self.figure.axes[0].ignore_existing_data_limits = True
+        self.figure.axes[0].autoscale_view()
+        self._format_cut_xabel()
+        for textobj in self.figure.findobj(text.Text):
+            textobj.set_fontsize(8)
+        self.figure.tight_layout()
+
+    def _format_cut_xabel(self):
+        xlab = self.figure.axes[0].get_xlabel()
+        istart = xlab.index('(')
+        iend = xlab.index(')')
+        xunit_str = xlab[iend + 1:].replace('Ang^-1', '$\\AA^{-1}$)').replace('(', '').replace(')', '')
+        xlab = xlab[0:istart] + xlab[istart:iend + 1].replace(' ', ', ') + xunit_str
+        self.figure.axes[0].set_xlabel(xlab)
+
+
+class CutRepresentation:
```

Comment:
I think this would ideally be in its own unit (similar to the peak representations in the peak viewer) - but see how much time you've got